lightning/src/ln/interactivetxs.rs

```diff
@@ -320,10 +320,7 @@ impl InteractiveTxSigningSession {
 	/// unsigned transaction.
 	pub fn received_tx_signatures(
 		&mut self, tx_signatures: TxSignatures,
-	) -> Result<(Option<TxSignatures>, Option<Transaction>), ()> {
-		if self.counterparty_sent_tx_signatures {
-			return Ok((None, None));
-		};
+	) -> Result<(Option<TxSignatures>, Transaction), ()> {
```

Comment:
Did `Transaction` never need to be an `Option`? If so, could you make a separate commit for this change?